src/sentry/utils/snowflake.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import OrderedDict
+
+from django.conf import settings
+
+
+@dataclass(frozen=True, eq=True)
+class SnowflakeBitSegment:
+    length: int
+    name: str
+
+    def validate(self, value):
+        if self.length <= 0:
+            raise Exception("The length should be a positive number")
+        if value >> self.length != 0:
+            raise Exception(f"{self.name} exceed max bit value of {self.length}")
+        return True
+
+
+SENTRY_EPOCH_START = datetime(2022, 4, 26, 0, 0).timestamp()
+ID_VALIDATOR = SnowflakeBitSegment(settings.SNOWFLAKE_ID_LENGTH, "Snowflake ID")
+
+VERSION_ID = SnowflakeBitSegment(settings.SNOWFLAKE_VERSION_ID_LENGTH, "Version ID")
+TIME_DIFFERENCE = SnowflakeBitSegment(settings.SNOWFLAKE_TIME_DIFFERENCE_LENGTH, "Time difference")
+REGION_ID = SnowflakeBitSegment(settings.SNOWFLAKE_REGION_ID_LENGTH, "Region ID")
+REGION_SEQUENCE = SnowflakeBitSegment(settings.SNOWFLAKE_REGION_SEQUENCE_LENGTH, "Region sequence")
+
+
+def snowflake_id_generation():
+    segment_values = OrderedDict()
+    segment_values[VERSION_ID] = 1
+    segment_values[TIME_DIFFERENCE] = 0
+    segment_values[REGION_ID] = 0
```

Comment:
@maxiuyuan Just out of curiosity, what's the plan for populating/setting the region id? Will there be `settings.REGION_ID`?

Reply:
We could set `settings.REGION_ID` but that doesn't seem necessary. The values of `REGION_ID` and `VERSION_ID` could just be hard coded within here